plugins/modules/nd_setup.py

```diff
@@ -150,6 +152,11 @@
         - The serial number of the node.
         type: str
         required: true
+      role:
+        description:
+        - The role of the node.
+        type: str
+        choices: [ primary, secondary, standby ]
```

Comment:
there specific information regarding the default role for nodes in the API documentation. the problem is that this feature was introduced in ND v3.1.1 as well as deployment mode but the role is required. I tried several test without the role being defined and it seems like it defaults to `primary`. I am gonna propose something but let me know what you think.

Reply:
the API works so that for previous ND version, new features introduced in later ones are ignored if used in the payload. So this should not be a problem.